src/main/java/umc/GrowIT/Server/repository/UserChallengeRepository.java

```diff
@@ -1,9 +1,45 @@
 package umc.GrowIT.Server.repository;
 
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import umc.GrowIT.Server.domain.Challenge;
 import umc.GrowIT.Server.domain.UserChallenge;
+import umc.GrowIT.Server.domain.enums.ChallengeType;
+
+import java.util.List;
+import java.util.Optional;
+
 import java.util.Optional;
 
 public interface UserChallengeRepository extends JpaRepository<UserChallenge, Long> {
-    Optional<UserChallenge> findByIdAndUserId (Long userChallengeId, Long userId);
+
+    // 특정 챌린지 ID와 매핑된 인증 내역 조회
+    @Query("SELECT uc FROM UserChallenge uc WHERE uc.challenge.id = :challengeId")
+    Optional<UserChallenge> findByChallengeId(@Param("challengeId") Long challengeId);
+    @Query("SELECT uc FROM UserChallenge uc WHERE uc.challenge.id = :challengeId AND uc.user.id = :userId")
+    Optional<UserChallenge> findByChallengeIdAndUserId(@Param("challengeId") Long challengeId, @Param("userId") Long userId);
+
+    // 유저별 완료한 챌린지 내역 조회
+    @Query("SELECT uc FROM UserChallenge uc " +
+            "WHERE uc.user.id = :userId AND uc.completed = true")
+    List<UserChallenge> findCompletedUserChallengesByUserId(@Param("userId") Long userId);
+
+    // 특정 유저의 인증 완료된 챌린지 조회
+    @Query("SELECT uc FROM UserChallenge uc " +
+            "WHERE uc.user.id = :userId " +
+            "AND (:status IS NULL OR uc.challenge.dtype = :status) " +
+            "AND uc.completed = true")
+    List<UserChallenge> findUserChallengesByStatusAndCompletion(
+            @Param("userId") Long userId,
+            @Param("status") ChallengeType challengeType,
+            @Param("completed") Boolean completed);
+
+    // 특정 유저가 완료하지 않은 챌린지 조회
+    @Query("SELECT c FROM Challenge c " +
+            "WHERE c.id NOT IN (" +
+            "  SELECT uc.challenge.id FROM UserChallenge uc WHERE uc.user.id = :userId AND uc.completed = true" +
+            ") " +
+            "AND (:status IS NULL OR c.dtype = :status)")
+    List<Challenge> findAvailableChallengesForUser(@Param("userId") Long userId, @Param("status") ChallengeType challengeType);
```

Comment:
```suggestion
List<UserChallenge> findByUserIdAndCompleted (Long userId,boolean completed);
```
이렇게 처리했을 때 문제가 있을까요?

Reply:
@syeongk 다른 API 먼저 구현하고 나중에 코드 리뷰 받은 부분들 수정해도 될까요? 피드백 참고하여 코드 하나하나 수정 중에 있는데 생각보다 오래 걸려서 다른 API 먼저 구현하는게 낫지않을까 생각해서요. 코드리뷰를 먼저 해결해야 한다면 어떻게든 해결..해보겠습니다

Reply:
기존 코드 리뷰드린 거 반영해서 다른 API 도 고려해서 만드는게 더 좋지 않을까요? 그리고 제가 리뷰 드린 부분에 있어서 의견있으시면 남겨주시고 피드백을 남겨주시면 좋을 것 같습니다.

Reply:
네 리뷰 주신거 참고하여 개발하고 있습니다만 pr 승인을 받지 않고 다른 기능을 개발해도 문제 없는건지, 또는 pr 승인 받고 나중에 디밸롭에서 따로 브랜치를 파서 수정해도 되는건지 궁금해서 질문드리는겁니다

Reply:
승현님 현재 이슈에서 문제되는 부분들 모두 해결하고 다른 기능을 구현하는게 맞지 않을까요? 그리고 PR 승인 받고 나중에 수정하는 건 상관없습니다.

Reply:
현재 스웨거에서 dtype(DAILY, RANDOM)으로 조회가 되지 않는 상황이라 먼저 해결하고 피드백 주신거 참고하여 수정하겠습니다!

Reply:
제가 제안해드린 코드에 문제가 있다는 말씀이신지 어떤 상황인지 정확히 얘기해주셔야 이해할 것 같습니다